### PR TITLE
driver-only ECC: BN.x509 testing

### LINKED ISSUE #7755: driver-only ECC: BN.x509 testing
In the `all.sh` component that was created in #7754, re-enable X.509.

Ensure parity and fix any issue that might arise.

**Depends on:** #7754, #7748 and #6843
Follow-up: #7756

### PR DESCRIPTION
**Pull request by valeriosetti**
This PR is the continuation of #7989 and it enables also X509 testing.

Depends on #7989 
Resolves #7755

## PR checklist

- [x] **changelog** will  be provided in #7757
- [x] **backport** not required because it's an enhancement
- [x] **tests** not required because the parity analysis element was already added in #7989

### CODE REVIEW

**Review by mpg — APPROVED** [review]
LGTM

**Comment by tom-daubney-arm** [comment]
There are conflicts in `pk.h`. Is this something that we expect to disappear once #7989 is merged?

**Comment by valeriosetti** [comment]
> There are conflicts in `pk.h`. Is this something that we expect to disappear once #7989 is merged?

I think so because it was something that I also saw this morning when I rebased #7989, so it does not belong to the code of this PR. BTW it's a minor conflict (moving things between `pk.h` and `build_info.h`), so you can still take a look at the PR and perhaps waiting for the rebase to do the final approval.

**Review by tom-daubney-arm — COMMENTED** [review]
This has been rebased as I was mid-way through and as a result I am not sure if my comments are relevant any more. If they are, can we please address them? They are only about the grammar in some comments, and the comment existed before this PR but I think it is worth correcting. Otherwise this PR LGTM

**Comment by valeriosetti** [comment]
> This has been rebased as I was mid-way through and as a result I am not sure if my comments are relevant any more. If they are, can we please address them?

Sure! I will go through all of them and apply the fixes you proposed ;)

File: include/mbedtls/build_info.h
Comment (valeriosetti) [reply include/mbedtls/build_info.h]: Thanks for the suggestion!
Pro tip for next time: while reviewing there is a "+/-" button on top of the comment box which is helpful (and also looks nicer) when you want to propose some typo fix ;)

Comment (tom-daubney-arm) [reply include/mbedtls/build_info.h]: Not sure where you mean? What is the comment box in this context?

File: include/mbedtls/pk.h
Comment (tom-daubney-arm) [inline include/mbedtls/pk.h:205]: Sorry, not trying to be extra picky but one thing was missed: "affect" --> "affects". It needs the "s" on the end.

(I've done this the old way since I don't understand the pro tip from the last round of review...

File: include/mbedtls/build_info.h
Comment (valeriosetti) [reply include/mbedtls/build_info.h]: When you write an answer (like this) or add comment during a review, there is a box in which you write your comment. On top of that box there are a few buttons like italic, bold, etc... Between these there is also "Suggestions" which have a symbol similar to "+/-"

Comment (tom-daubney-arm) [reply include/mbedtls/build_info.h]: Yeah nice, I see it now, will try that next time.

**Review by tom-daubney-arm — DISMISSED** [review]
LGTM, thanks for fixing the existing grammar.

**Review by mpg — APPROVED** [review]
LGTM (the rebase and the new commit).

**Review by tom-daubney-arm — APPROVED** [review]

**Pull request merged**
